tests/foreman/ui/test_computeresource_gce.py

```diff
@@ -245,6 +246,14 @@ def test_positive_gce_cloudinit_provision_end_to_end(
 
     :expectedresults: Host is provisioned successfully
     """
+
+    @request.addfinalizer
+    def _finalize():
+        gcehost = sat_gce.api.Host().search(query={'search': f'name={hostname}'})
+        if gcehost:
+            gcehost[0].delete()
+        googleclient.disconnect()
```

Comment:
Can we move it at least below after variables get defined? functionally I think it wouldn't affect as we're checking if host exists here first before delete